### PR TITLE
Standardize linting file and fix or ignore lots of linting errors

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: authority/provisioner/claims.go [hunk 1]
```diff
@@ -38,7 +38,8 @@ type Claimer struct {
 // NewClaimer initializes a new claimer with the given claims.
 func NewClaimer(claims *Claims, global Claims) (*Claimer, error) {
 	c := &Claimer{global: global, claims: claims}
-	return c, c.Validate()
+	err := c.Validate()
+	return c, err
```
[the remark this answers is not in the document]
Comment (maraino) [reply authority/provisioner/claims.go:42]: If we really want this linter, we probably should do
```suggestion
	if err := c.Validate(); err != nil {
		return nil, err
	}
	return c, nil
```

File: cas/apiv1/services.go [hunk 2]
```diff
@@ -59,9 +59,9 @@ func (t Type) String() string {
 	return strings.ToLower(string(t))
 }
 
-// ErrNotImplemented is the type of error returned if an operation is not
-// implemented.
-type ErrNotImplemented struct {
+// ErrNotImplemented is the type of error returned if an operation is not implemented.
+type ErrNotImplemented struct { //nolint:errname // ignore error name warning
```
Comment (maraino) [inline cas/apiv1/services.go:63]: We should rename it.

**Review by maraino — CHANGES_REQUESTED** [review]
Let's try to make this As-like methods internal or use types and errors.As.

File: api/render/render.go [hunk 3]
```diff
@@ -72,12 +72,22 @@ type RenderableError interface {
...
 
+// AsRenderableError attempts to return an error type that implements the
+// RenderableError interface.
+func AsRenderableError(err error) (RenderableError, bool) {
+	//nolint:errorlint // ignore type assertion warning. casting to interface is hard.
+	if r, ok := err.(RenderableError); ok {
+		return r, true
+	}
+	return nil, false
+}
```
[the remark this answers is not in the document]
Comment (dopey) [reply api/render/render.go:83]: I'm not entirely sure how I would do this. Even if I define a RenderableError type struct {}, how would I handle potential errors from outside packages. Those packages definitely won't be using our RenderableError type. Like here: https://github.com/smallstep/certificates/blob/b3b37400264d9e9200f0c49479501efe5fb6598b/errs/error.go#L257.

**Pull request closed without merging**